crates/sparrow-interfaces/src/source.rs

```diff
@@ -0,0 +1,12 @@
+use arrow_schema::SchemaRef;
+use futures::stream::BoxStream;
+use sparrow_batch::Batch;
+use sparrow_core::ReadConfig;
+
+use crate::Error;
+
+/// Defines how a source provides data to the execution layer.
```

Comment:
I think that's just the `read` method. Probably want a different comment on the trait, and then move this to the `read` method (with some more details)?